src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -2,75 +2,34 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-use std::{
-    fmt,
-    io::{self, Read, Seek, SeekFrom},
-    str::from_utf8,
-};
+use std::io::{Read, Seek};
 
-use byteorder::{ByteOrder, LittleEndian};
 use chrono::{DateTime, Utc};
-use crc::crc32;
 use uuid::Uuid;
 
-use devicemapper::{Sectors, IEC, SECTOR_SIZE};
-
 use crate::{
     engine::{
         strat_engine::{
             backstore::metadata::{
                 mda,
-                sizes::{
-                    static_header_size, BDAExtendedSize, BlockdevSize, MDADataSize, MDASize,
-                    ReservedSize, STATIC_HEADER_SIZE,
-                },
+                sizes::{BDAExtendedSize, BlockdevSize, MDADataSize, STATIC_HEADER_SIZE},
+                static_header::StaticHeader,
             },
             device::SyncAll,
         },
         types::{DevUuid, PoolUuid},
     },
-    stratis::{ErrorEnum, StratisError, StratisResult},
+    stratis::StratisResult,
 };
 
-const RESERVED_SECTORS: Sectors = Sectors(3 * IEC::Mi / (SECTOR_SIZE as u64)); // = 3 MiB
-
-const STRAT_MAGIC: &[u8] = b"!Stra0tis\x86\xff\x02^\x41rh";
-
-const STRAT_SIGBLOCK_VERSION: u8 = 1;
-
-/// Get a Stratis pool UUID and device UUID from any device.
-/// If there is an error while obtaining these values return the error.
-/// If the device does not appear to be a Stratis device, return None.
-pub fn device_identifiers<F>(f: &mut F) -> StratisResult<Option<(PoolUuid, DevUuid)>>
-where
-    F: Read + Seek + SyncAll,
-{
-    StaticHeader::setup(f).map(|sh| sh.map(|sh| (sh.pool_uuid, sh.dev_uuid)))
-}
-
-/// Remove Stratis identifying information from device.
-pub fn disown_device<F>(f: &mut F) -> StratisResult<()>
-where
-    F: Seek + SyncAll,
-{
-    StaticHeader::wipe(f)
-}
-
-// Transform a constant in sectors to a constant in bytes
-macro_rules! bytes {
-    ($number:expr) => {
-        $number * devicemapper::SECTOR_SIZE
-    };
-}
-
 #[derive(Debug)]
 pub struct BDA {
     header: StaticHeader,
     regions: mda::MDARegions,
 }
 
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
-enum MetadataLocation {
+pub enum MetadataLocation {
```

Comment:
Please move the definition of this type into the ```static_header``` module and import it into the bda module. It defines the API of some ```static_header``` methods which some bda methods call, so that is the best place to put it.